File: tests/filetypes/sample.swift
// Swift example to test editor settings

import Foundation

// Define a variable and a constant
var name: String = "Swift"
let version: Double = 5.5

// Print a message
print("Hello, \(name)! Version: \(version)")

// Function to add two numbers
func add(_ a: Int, _ b: Int) -> Int {
    return a + b
}

// Call the function
let result = add(5, 10)
print("Sum: \(result)")

// Conditional statement
if result > 10 {
    print("Result is greater than 10")
} else {
    print("Result is less than or equal to 10")
}

// Arrays and loops
let numbers = [1, 2, 3, 4, 5]
for number in numbers {
    print("Number: \(number)")
}

// Dictionary
let person: [String: Any] = [
    "name": "Alice",
    "age": 30,
    "email": "alice@example.com"
]
print("Person: \(person["name"] ?? "") is \(person["age"] ?? 0) years old.")

// Classes and objects
class Person {
    var name: String
    var age: Int
    
    init(name: String, age: Int) {
        self.name = name
        self.age = age
    }
    
    func greet() -> String {
        return "Hello, my name is \(name) and I am \(age) years old."
    }
}

let person1 = Person(name: "Bob", age: 35)
print(person1.greet())

// Optionals and optional binding
var optionalString: String? = "This is an optional"
if let unwrappedString = optionalString {
    print("Unwrapped string: \(unwrappedString)")
} else {
    print("optionalString is nil")
}

// Closures
let square = { (x: Int) -> Int in
    return x * x
}
print("Square of 4: \(square(4))")

// Error handling
enum NetworkError: Error {
    case badURL
    case noData
}

func fetchData(from url: String) throws {
    if url.isEmpty {
        throw NetworkError.badURL
    }
    print("Data fetched from \(url)")
}

do {
    try fetchData(from: "")
} catch NetworkError.badURL {
    print("Bad URL provided.")
} catch {
    print("An unknown error occurred.")
}

File: tests/filetypes/sample.fs
// F# example to test editor settings

// Variables and constants
let name = "F#"
let version = 5.0

// Print a message
printfn "Hello, %s! Version: %.1f" name version

// Function to add two numbers
let add a b = a + b

// Call the function
let result = add 5 10
printfn "Sum of 5 and 10: %d" result

// Conditional statement
if result > 10 then
    printfn "Result is greater than 10"
else
    printfn "Result is less than or equal to 10"

// Lists and loops
let numbers = [1; 2; 3; 4; 5]
for num in numbers do
    printfn "Number: %d" num

// Maps (similar to dictionaries)
let person = Map.ofList [("name", "Alice"); ("age", "30"); ("email", "alice@example.com")]
printfn "%s is %s years old." (person.["name"]) (person.["age"])

// Define a simple record type
type Person = { Name: string; Age: int }

let person1 = { Name = "Bob"; Age = 35 }
printfn "Hello, my name is %s and I am %d years old." person1.Name person1.Age

// Recursive function to calculate factorial
let rec factorial n =
    if n = 0 then 1
    else n * factorial (n - 1)

printfn "Factorial of 5: %d" (factorial 5)

// Lambda expression
let square = fun x -> x * x
printfn "Square of 4: %d" (square 4)

// Pattern matching
let processMessage message =
    match message with
    | "hello" -> printfn "Received a hello message"
    | "goodbye" -> printfn "Received a goodbye message"
    | _ -> printfn "Unknown message"

processMessage "hello"

// Try-catch for exception handling
try
    let fetchData url =
        if url = "" then failwith "URL cannot be empty"
        printfn "Fetching data from %s" url
    fetchData ""
with
| ex -> printfn "Error: %s" ex.Message

